apps/src/StudioApp.js

```diff
@@ -645,6 +697,53 @@ StudioApp.prototype.handleSharing_ = function (options) {
   }
 };
 
+StudioApp.prototype.renderShareFooter_ = function(container) {
+  var footerDiv = document.createElement('div');
+  footerDiv.setAttribute('id', 'footerDiv');
+  container.appendChild(footerDiv);
+
+  var reactProps = {
+    i18nDropdown: '',
+    copyrightInBase: false,
+    copyrightStrings: copyrightStrings,
+    baseMoreMenuString: window.dashboard.i18n.t('footer.built_on_code_studio'),
+    baseStyle: {
+      paddingLeft: 0,
+      width: $("#visualization").width(),
+    },
+    className: 'dark',
+    menuItems: [
+      {
+        text: window.dashboard.i18n.t('footer.try_hour_of_code'),
+        link: 'https://code.org/learn',
+        newWindow: true
+      },
+      {
+        text: window.dashboard.i18n.t('footer.how_it_works'),
+        link: location.href + "/edit",
+        newWindow: false
+      },
+      {
+        text: window.dashboard.i18n.t('footer.copyright'),
+        link: '#',
+        copyright: true
+      },
+      {
+        text: window.dashboard.i18n.t('footer.tos'),
+        link: "https://code.org/tos",
+        newWindow: true
+      },
+      {
+        text: window.dashboard.i18n.t('footer.privacy'),
+        link: "https://code.org/privacy",
+        newWindow: true
+      }
+    ]
+  };
+
+  window.dashboard.footer.render(React, reactProps, footerDiv);
```

Comment:
Agreed. I consolidated the copyright strings so that's only constructed in one place. Otherwise like you said the 3 different small footers (normal, applab, and this legacy share) have variations in their properties. I'm open to suggestions on consolidating it further.